### PR TITLE
simple prior use test

### PR DESCRIPTION
**Pull request by Affie**
I feel this test should pass

### CODE REVIEW

**Comment by dehann** [comment]
Hi, oh great thanks will resolve today!

**Comment by dehann** [comment]
hi, didnt get through this today, but will look first thing in the morning.

File: test/priorusetest.jl [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        tree, smt, hist = solveTree!(fg)
+        x0_m = getKDEMean(getKDE(getVariable(fg, :x0)))[1]
+        x1_m = getKDEMean(getKDE(getVariable(fg, :x1)))[1]
+        x2_m = getKDEMean(getKDE(getVariable(fg, :x2)))[1]
+
+        @info ("Testing means = 0 with 2 priors:\ngraphinit=$graphinit\nMeans: x0: $(x0_m), x1: $x1_m, x2: $x2_m")
+
+        @test isapprox(x0_m, 0.0, atol = 0.1)
+        @test isapprox(x1_m, 0.0, atol = 0.1)
+        @test isapprox(x2_m, 0.0, atol = 0.1)
```
Comment (dehann) [inline test/priorusetest.jl]: @Affie, first question: if you are putting two priors at the ends of a 3 variable chain `(x0,x1,x2)`, either at + or - 1 -- why would you expect them to all be at average 0?  I would expect them to be at + or 1 one respectively, with `x1` floating somewhere 0 in the middle?

This was based on me reading the code.

**Comment by dehann** [comment]
## first test graph

Screenshot from 2019-12-04 20-07-58

**Comment by Affie** [comment]
I don't agree on changes.
What about this: changing the variable ordering changes the answers.
image

GraphInit = true
Means: x0: 0.031117603087722025, x1: 0.1416983152637742, x2: 0.23945144191614984
Means: x0: -0.41846814644201763, x1: -0.3480670649963539, x2: -0.1642135339004435

GraphInit = false
Means: x0: 0.07968446871195635, x1: 0.14465511208606313, x2: 0.22910276233886342
Means: x0: -0.33884567700033713, x1: -0.281886695060186, x2: -0.19827585825173885

**Comment by Affie** [comment]
If #221 is the issue I brought up it is related.
And second graph tests for #458

File: test/priorusetest.jl (hunk 1 above)
Comment (dehann) [reply test/priorusetest.jl]: x0 and x2 should be near 0, and only x1 should be "on" 0.  Test with loose `atol=0.1` bit hacky, but okay :-)  x0 and x2 should be around 0.1 or something like that.

**Comment by dehann** [comment]
adding @tonioteran here for interest sake.

I strongly agree that changing the variable ordering should not change the numerical result.  Thanks for describing, I definitely need to spend time on this and make sure we understand what is happening here.

File: test/priorusetest.jl (hunk 1 above)
Comment (Affie) [reply test/priorusetest.jl]: Please note, I changed the standard deviation to 0.01.
So its more like -0.01, 0, 0.01 for x0,x1,x2.

-1/1 <-x0-> 1/0.01 <-x1-> 1/0.01 <-x2-> 1/1

**Comment by Affie** [comment]
I looks like it either has to do with the structure or the priors on the down solve.

File: test/priorusetest.jl (hunk 1 above)
Comment (dehann) [reply test/priorusetest.jl]: i'm assuming you did the calculation for multiplying two Gaussians, and found that the new mean should be at +- 0.01?  What I mean is if you were to multiply N(-1,0.1) * N(+1,0.1), the answer will be at zero with a new covariance slightly smaller than 0.1.

**Comment by dehann** [comment]
acknowledged thanks, definitely will do a deep dive on this soon.

**Comment by dehann** [comment]
xref #462

**Comment by dehann** [comment]
xref #485

**Comment by dehann** [comment]
xref #487

**Pull request merged**